src/platform/android/java/chip/platform/NsdManagerServiceResolver.java

```diff
@@ -138,20 +138,13 @@ public void publish(
     serviceInfo.setServiceName(serviceName);
 
     /**
-     * Note, subtypes registration is using an undocumented feature of android dns-sd
-     * service/mDNSResponder which MAY STOP WORKING in future Android versions. Here, set type =
-     * "${type},${subtypes1},${subtypes2},...", then subtypes1, subtypes2 etc are all registered to
-     * this dns-sd server, we can usd `dns-sd -B ${type},${subtypes}` or avahi-browse
-     * ${subtypes}._sub.${type} -r to browser it
+     * TODO: Add subtypes registration when the Android NsdManager provides a documented mechanism
+     * to publish them. See https://issuetracker.google.com/u/3/issues/314256875
      */
-    StringBuilder sb = new StringBuilder(type);
-    for (String subType : subTypes) {
-      sb.append(",").append(subType);
-    }
-    serviceInfo.setServiceType(sb.toString());
+    serviceInfo.setServiceType(type);
```

Comment:
This regresses on previous android versions it seems (based on comments).

Is this true? what is the sideffect for this? could we add version checks here to not change behavior?

Reply:
#15009 seems to have added this on purpose.

@xylophone21 do you have any feedback here? what would this change?

Reply:
Looks like somebody submitted a fix for the original problem in AOSP: [https://android-review.googlesource.com/c/platform/packages/modules/Connectivity/+/2867134](https://android-review.googlesource.com/c/platform/packages/modules/Connectivity/+/2867134)

Closing this PR as not required.